odxtools/singleecujob.py

```diff
@@ -137,12 +137,18 @@ def from_et(et_element: Element, doc_frags: List[OdxDocFragment]) -> "SingleEcuJ
             for el in et_element.iterfind("NEG-OUTPUT-PARAMS/NEG-OUTPUT-PARAM")
         ]
 
+        sdgs = create_sdgs_from_et(et_element.find("SDGS"), doc_frags)
+
         # Read boolean flags. Note that the "else" clause contains the default value.
         is_mandatory_raw = odxstr_to_bool(et_element.get("IS-MANDATORY"))
         is_executable_raw = odxstr_to_bool(et_element.get("IS-MANDATORY"))
         is_final_raw = odxstr_to_bool(et_element.get("IS-FINAL"))
 
-        sdgs = create_sdgs_from_et(et_element.find("SDGS"), doc_frags)
+        diag_class_type: Optional[DiagClassType] = None
+        if (diag_class_type_str := et_element.get("DIAGNOSTIC-CLASS")) is not None:
+            if diag_class_type_str not in DiagClassType.__members__:
+                odxraise(f"Encountered unknown diagnostic class type '{diag_class_type_str}'")
+            diag_class_type = DiagClassType(diag_class_type_str)
```

Comment:
`DiagClassType(diag_class_type_str)` will throw even if `odxraise` does not

Reply:
yes, but when using the non-strict mode, all bets are off: In non-strict mode there are plenty of instances where methods are called for objects that can be `None` in principle (e.g., [here](https://github.com/mercedes-benz/odxtools/blob/main/odxtools/statetransition.py#L45-L46))...